sirepo/auth/ldap.py

```diff
@@ -0,0 +1,56 @@
+"""LDAP login
+:copyright: Copyright (c) 2018-2022 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern.pkdebug import pkdlog, pkdp
+from pykern.pkcollections import PKDict
+import ldap
+import sirepo.quest
+
+AUTH_METHOD = "ldap"
+
+AUTH_METHOD_VISIBLE = True
+
+_cfg = None
+
+class API(sirepo.quest.API):
+    @sirepo.quest.Spec("require_cookie_sentinel")
+    def api_authLdapAuthorized(self):
```

Comment:
I don't think this is "authorized". This is not like oauth. Rather, it's more like authEmailLogin. This does the authorization, but it's all in one function, from the GUI perspective it is asking to login.

Reply:
Changing to authEmailLogin.

Would it be useful to seperate the authorization into its own method?

Reply:
It's not "authEmailLogin" It's authLdapLogin.